aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -119,4 +139,95 @@ private boolean shouldRemoveFromGlobalCluster(
         return StringUtils.hasValue(previousResourceState.getGlobalClusterIdentifier()) &&
                 StringUtils.isNullOrEmpty(desiredResourceState.getGlobalClusterIdentifier());
     }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> rebootCluster(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final ResourceModel resourceModel
+    ) {
+        DBCluster cluster;
+        try {
+            cluster = fetchDBCluster(proxyClient, resourceModel);
+        } catch (Exception ex) {
+            return Commons.handleException(progress, ex, DEFAULT_DB_CLUSTER_ERROR_RULE_SET);
+        }
+
+        ProgressEvent<ResourceModel, CallbackContext> result = progress;
+
+        for (final DBClusterMember dbClusterMember: cluster.dbClusterMembers()) {
+            final ProgressEvent<ResourceModel, CallbackContext> p = proxy.initiate("rds::reboot-dbcluster-member-" + dbClusterMember.dbInstanceIdentifier(),
+                            proxyClient,
+                            progress.getResourceModel(),
+                            progress.getCallbackContext())
+                    .translateToServiceRequest(model -> Translator.rebootDbInstanceRequest(dbClusterMember.dbInstanceIdentifier()))
+                    .backoffDelay(config.getBackoff())
+                    .makeServiceCall((rebootDbInstanceRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
+                            rebootDbInstanceRequest,
+                            proxyInvocation.client()::rebootDBInstance
+                    ))
+                    .stabilize((rebootDbInstanceRequest, rebootDbInstanceResponse, proxyInvocation, model, context) ->
+                            isDBInstanceStabilized(proxyClient, dbClusterMember.dbInstanceIdentifier(), model))
+                    .handleError((createRequest, exception, client, model, callbackCtx) -> Commons.handleException(
+                            ProgressEvent.progress(model, callbackCtx),
+                            exception,
+                            DEFAULT_DB_CLUSTER_ERROR_RULE_SET
+                    ))
+                    .progress();
+
+            if (p.isFailed() || p.isInProgressCallbackDelay()) {
+                return p;
+            }
+            result = p;
+        }
+        return result;
```

Comment:
Not sure I understand chaining in this case. `result` is always set to the newly initiated proxy invocation on line 159. Shall we chain them up with `then()` invocations?

Reply:
It is possible to do that through chaining, yes. The current logic is that this function reboots one instance at the time: stabilization code returns in-progress `ProgressEvent` with correct callback context. This context is used in subsequent invocations.